pages/index.tsx

```diff
@@ -95,39 +93,56 @@ function KeyboardComponent() {
   let pressedKeysStr = pressedKeysArr.join(" ");
 
   return (
-    <Keyboard
-      keyboardRef={(r) => (keyboard.current = r)}
-      layoutName={layout}
-      onKeyPress={(button) => onKeyPress(button)}
-      layout={{
-        default: [
-          "` 1 2 3 4 5 6 7 8 9 0 - = {bksp}",
-          "{tab} q w e r t y u i o p [ ] \\",
-          "{lock} a s d f g h j k l ; ' {enter}",
-          "{shift} z x c v b n m , . / {shift}",
-          ".com @ {space}",
-        ],
-        shift: [
-          "~ ! @ # $ % ^ & * ( ) _ + {bksp}",
-          "{tab} Q W E R T Y U I O P { } |",
-          '{lock} A S D F G H J K L : " {enter}',
-          "{shift} Z X C V B N M < > ? {shift}",
-          ".com @ {space}",
-        ],
+    <ThemeContext.Consumer>
+      {(theme) => {
+        return (
+          <Keyboard
+            keyboardRef={(r) => (keyboard.current = r)}
+            layoutName={layout}
+            theme={
+              theme.name === "portGore"
+                ? `hg-theme-default ${styles.portGore}`
+                : `${styles.solitare} hg-theme-default`
+            }
+            onKeyPress={(button) => onKeyPress(button)}
+            layout={{
+              default: [
+                "` 1 2 3 4 5 6 7 8 9 0 - = {bksp}",
+                "{tab} q w e r t y u i o p [ ] \\",
+                "{lock} a s d f g h j k l ; ' {enter}",
+                "{shift} z x c v b n m , . / {shift}",
+                ".com @ {space}",
+              ],
+              shift: [
+                "~ ! @ # $ % ^ & * ( ) _ + {bksp}",
+                "{tab} Q W E R T Y U I O P { } |",
+                '{lock} A S D F G H J K L : " {enter}',
+                "{shift} Z X C V B N M < > ? {shift}",
+                ".com @ {space}",
+              ],
+            }}
+            buttonTheme={[
+              {
+                class:
+                  theme.name === "portGore"
+                    ? `${styles["pressed-key-portGore"]}`
+                    : `${styles["pressed-key-solitare"]}`,
+                buttons: pressedKeysStr ? pressedKeysStr : "empty",
+                // Placeholder value needed, otherwise ESLINT error
+              },
+              {
+                class:
+                  theme.name === "portGore"
+                    ? `${styles["currently-pressed-portGore"]}`
+                    : `${styles["currently-pressed-solitare"]}`,
+                buttons: currentlyPressedKeys ? currentlyPressedKeys : "empty",
+                // Placeholder value needed, otherwise ESLINT error
+              },
```

Comment:
You can add the button styles to the theme interface, so that you can abstract the theme selection logic and use the theme interface right away dynamically, and hence you wouldn't need to have if/else statements for the themes anywhere else in the code. 
The advantage for having that implementation would be to allow us to easily add extra themes in the future without having to alter anything other than the theme interface. The components would remain untouched, since we are only exposing the selected theme interface.

maybe @reasonablytall could have an opinion about that as well.